modules/core/src/main/java/org/apache/ignite/internal/util/ByteUtils.java

```diff
@@ -123,12 +123,12 @@ public static byte[] toBytes(Object obj) {
      * @param bytes Byte array.
      * @return Object.
      */
-    public static Object fromBytes(byte[] bytes) {
+    public static <T> T fromBytes(byte[] bytes) {
         try (
                 var bis = new ByteArrayInputStream(bytes);
                 var in = new ObjectInputStream(bis)
         ) {
-            return in.readObject();
+            return (T) in.readObject();
```

Comment:
This is an unsafe cast, I would prefer to leave the current approach with explicit casting

Reply:
Well it's less code and the runtime implications are the same, are they not?

Reply:
They are not exactly the same, because with the explicit cast the ClassCastException may be thrown earlier

Reply:
I'm against such code, but technically, both approaches are absolutely the same here. This method will return Object without cast, because no type information is available to it. It's all just sugar.

Reply:
ok, I did a little bit of thinking and this code looks safe and is more convenient to use. Please ignore my comment